src/engine/Algebra.ts

```diff
@@ -68,11 +71,25 @@ module ex {
          return true;
       }
 
+      constructor();
+      /**
+       * @param vectorOptions coordindates of the Vector
+       */
+      constructor(vectorOptions: IVectorOptions);
       /**
        * @param x  X component of the Vector
        * @param y  Y component of the Vector
        */
-      constructor(public x: number, public y: number) { }
+      constructor(x: number, y: number);
+      constructor(optionsOrX?: number | IVectorOptions, y?: number) {
+            if (typeof optionsOrX === 'number') {
```

Comment:
e.g. if a constructor currently takes an Actor as it's only parameter we wouldn't make an options bag constructors consisting of just `{ actor: new Actor() }`. This also adheres to my previous suggestion of just doing this for Params > 2. Cool.